### PR TITLE
Add support for mobile devices, integrate sentry

### PR DESCRIPTION
**Pull request by skusnierz**

### CODE REVIEW

File: template/src/hooks/useWindowSize.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+};
+
+const useWindowSize = (): WindowSize => {
+  const [size, setSize] = useState<WindowSize>({} as WindowSize);
+
+  useLayoutEffect(() => {
+    function updateSize() {
+      setSize({ width: window.innerWidth, height: window.innerHeight });
+    }
+    window.addEventListener('resize', updateSize);
```
Comment (rafalkociatkiewicz) [inline template/src/hooks/useWindowSize.ts:15]: add debounce

File: template/src/pages/login/components/Form.tsx [hunk 2]
```diff
@@ -55,35 +59,57 @@ const LoginForm = () => {
...
-        {t('loginPage.submit')}
+        {t('pages.loginPage.components.form.submit')}
       </StyledButton>
     </Form>
   );
 };
 
 const StyledButton = styled(Button)`
-  width: 400px;
+  @media ${getMediaQueryBreakpoint('xs')} {
```
Comment (rafalkociatkiewicz) [inline template/src/pages/login/components/Form.tsx:74]: const baseSizesCzyCos = css` XYZ `;

const StyledButton = styled(Button)`
${baseSizesCzyCos}
`;

File: template/src/templates/main/components/mobile/Navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { TabBar } from 'antd-mobile';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { MenuItemType } from '../..';
+
+interface NavbarProps {
```
Comment (rafalkociatkiewicz) [inline template/src/templates/main/components/mobile/Navbar.tsx:6]: wydzielić to aby można było użyć w Menu, SideMenu, AnyOtherMenu

File: template/src/theme/globalStyle.ts [hunk 4]
```diff
@@ -28,12 +28,37 @@ const GlobalStyle = createGlobalStyle`
...
+            height: ${({ theme: { spacing } }) =>
+              spacing.spacing16 + spacing.spacing4}px;
+            width: ${({ theme: { spacing } }) =>
+              spacing.spacing16 + spacing.spacing4}px;
+        }
+    }
+
+    .ant-menu:not(.ant-menu-horizontal) .ant-menu-item-selected {
+        background-color: ${getColor('primaryHover')} !important;
+        color: ${getColor('text01')} !important;;
```
Comment (rafalkociatkiewicz) [inline template/src/theme/globalStyle.ts:51]: ';;' why?

**Review by rafalkociatkiewicz — CHANGES_REQUESTED** [review]
Have fun fixing that

**Review by rafalkociatkiewicz — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -25,6 +25,52 @@ This package is a template that let's you generate an example boilerplate app wi
...
+## Folder structure
+
+```
+cra-template-crustlab
+└───src
+│   │   App.tsx
+│   │   App.css
+│   └───assets
+│   	└───images  all images must be placed here
+|	└───components	reusable common components go here
```
Comment (spiewak) [inline README.md:37]: I believe that it's a problem with formatting but `components`, `contexts` and `ThemeContext.tsx` should be considered as assets.

File: README.md [hunk 6]
```diff
@@ -25,6 +25,52 @@ This package is a template that let's you generate an example boilerplate app wi
...
+|	└───contexts  all context providers go here
+|       └─── ThemeContext.tsx
+|   └───forms   all files related to form validation go here
+|        └───login
+|            └───useLoginForm.ts
+|	└───helpers reusable common utility functions/libs go here
```
Comment (spiewak) [inline README.md:43]: Another problem with formatting `forms` should not contain `pages`, `home`, `services` etc.

File: README.md [hunk 7]
```diff
@@ -42,6 +88,18 @@ Above command creates **lang/en.json** file with english messages. Compare newly
 1. Create a new file e.g. **fr.json** in **lang** directory.
 2. Modify configure **i18n.ts** file, import new translation file and add to resources array.
 
+## Sentry
+
+Sentry is a developer-first error tracking and performance monitoring platform that helps developers see what actually matters, solve quicker, and learn continuously about their applications.
+Sentry configuration is done in **index.tsx** the file, Sentry is only run on a production environment.
```
Comment (spiewak) [inline README.md:94]: Sentry configuration is done in **index.tsx** file. Sentry is run only in a production environment.

File: README.md [hunk 8]
```diff
@@ -42,6 +88,18 @@ Above command creates **lang/en.json** file with english messages. Compare newly
...
+In order to configure the sentry correctly, we need to define two env variables **REACT_APP_SENTRY_DSN** and **REACT_APP_SETNRY_AUTH_TOKEN** and put them in our **.env** configuration file.
+To create a new auth token (**REACT_APP_SETNRY_AUTH_TOKEN**), or simply copy an existing one, just open sentry.io in your browser and then go to **Settings > Account > API > Auth Tokens**.
+In order for sentry to correctly display the error stack trace, we need to uplod the source map. To do this, we can use a previously prepared script **sentryCreateRelease.js** and run this npm command :
```
Comment (spiewak) [inline README.md:97]: In order to correctly display the error stack trace, we need to upload the source map. To do this, we can use a previously prepared script **sentryCreateRelease.js** and run this npm command :

File: README.md (hunk 5 above)
Comment (skusnierz) [reply README.md]: Fixed

File: README.md (hunk 6 above)
Comment (skusnierz) [reply README.md]: Fixed

**Comment by spiewak** [comment]
@skusnierz If there are no more comments, merge please :-)

**Pull request merged**